src/main/scala/com/microsoft/partnercatalyst/fortis/spark/TransformManager.scala

```diff
@@ -0,0 +1,256 @@
+    if (transformContext == null) {
+      transformContext = TransformContext()
+    }
+
+    transformContext = transformContext.copy(
+      siteSettings = delta.siteSettings.getOrElse(transformContext.siteSettings),
+      langToWatchlist = delta.langToWatchlist match {
+        case Some(list) => sparkContext.broadcast(list)
+        case None => transformContext.langToWatchlist
+      },
+      blacklist = delta.blacklist match {
+        case Some(list) => sparkContext.broadcast(list)
+        case None => transformContext.blacklist
+      },
+      locationsExtractorFactory = delta.locationsExtractorFactory match {
+        case Some(factory) => sparkContext.broadcast(factory)
+        case None => transformContext.locationsExtractorFactory
+      },
+      imageAnalyzer = delta.imageAnalyzer.getOrElse(transformContext.imageAnalyzer),
+      languageDetector = delta.languageDetector.getOrElse(transformContext.languageDetector),
+      sentimentDetectorAuth = delta.sentimentDetectorAuth.getOrElse(transformContext.sentimentDetectorAuth)
+    )
+  }
+
+  /**
+    * Builds a delta against the current transform context using the provided site settings.
+    * @param delta An existing delta to be used as a base.
+    * @return
+    */
+  private def deltaWithSettings(siteSettings: SiteSettings, delta: Delta = Delta()): Delta = {
+    // Note that parameters are call-by-name
+    def updatedField[T](isDirty: => Boolean, newVal: => T): Option[T] = {
+      // Invoke 'isDirty' only if transform context is not null
+      if (transformContext == null || isDirty)
+        Some(newVal)
+      else
+        None
+    }
+
+    delta.copy(
+      siteSettings = Some(siteSettings),
+      locationsExtractorFactory = updatedField(
+        siteSettings.geofence != transformContext.siteSettings.geofence,
+        new LocationsExtractorFactory(featureServiceClient, siteSettings.geofence).buildLookup()
+      ),
+      imageAnalyzer = updatedField(
+        siteSettings.cogVisionSvcToken != transformContext.siteSettings.cogVisionSvcToken,
+        new ImageAnalyzer(ImageAnalysisAuth(siteSettings.cogVisionSvcToken), featureServiceClient)
+      ),
+      languageDetector = updatedField(
+        siteSettings.translationSvcToken != transformContext.siteSettings.translationSvcToken,
+        new LanguageDetector(LanguageDetectorAuth(siteSettings.translationSvcToken))
+      ),
+      sentimentDetectorAuth = updatedField(
+        siteSettings.translationSvcToken != transformContext.siteSettings.translationSvcToken,
+        SentimentDetectorAuth(siteSettings.translationSvcToken)
+      )
+    )
+  }
+
+  /**
+    * Builds a delta against the current transform context using the provided watchlist.
+    * @param delta An existing delta to be used as a base.
+    * @return
+    */
+  private def deltaWithWatchlist(langToWatchlist: Map[String, List[String]], delta: Delta = Delta()) = {
+    delta.copy(
+      langToWatchlist = Some(langToWatchlist)
```

Comment:
That'd be a nice improvement that we could easily add on top of the current implementation in a future changeset. For now, I've favored the atomicity of fetching the entire set at a time given the current state of the Service Bus client library.